pkg/actuators/machine/actuator_test.go

```diff
@@ -3,184 +3,205 @@ package machine
+			event:               "aws-actuator-testing-machine: failed to create scope for machine: failed to create aws client: AWS client error",
+			invalidMachineScope: true,
+			awsError:            false,
 		},
 		{
-			name:    "Create machine event failed, reconciler's create failed",
-			machine: machine,
+			name: "Create machine event failed, reconciler's create failed",
 			operation: func(actuator *Actuator, machine *machinev1.Machine) {
 				actuator.Create(context.TODO(), machine)
 			},
-			event:    "Warning FailedCreate unable to remove stopped machines: error getting stopped instances: error",
-			awsError: true,
+			event:               "aws-actuator-testing-machine: reconciler failed to Create machine: unable to remove stopped machines: error getting stopped instances: AWS error",
+			invalidMachineScope: false,
+			awsError:            true,
 		},
 		{
-			name:    "Create machine event succeed",
-			machine: machine,
+			name: "Create machine event succeed",
 			operation: func(actuator *Actuator, machine *machinev1.Machine) {
 				actuator.Create(context.TODO(), machine)
 			},
-			event: "Normal Create Created Machine aws-actuator-testing-machine",
+			event:               "Created Machine aws-actuator-testing-machine",
+			invalidMachineScope: false,
+			awsError:            false,
 		},
 		{
-			name:    "Update machine event failed on invalid machine scope",
-			machine: machineInvalidProviderConfig,
+			name: "Update machine event failed on invalid machine scope",
 			operation: func(actuator *Actuator, machine *machinev1.Machine) {
 				actuator.Update(context.TODO(), machine)
 			},
-			event: "Warning FailedUpdate aws-actuator-testing-machine: failed to create scope for machine: failed to get machine config: error unmarshalling providerSpec: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal string into Go value of type v1beta1.AWSMachineProviderConfig",
+			event:               "aws-actuator-testing-machine: failed to create scope for machine: failed to create aws client: AWS client error",
+			invalidMachineScope: true,
+			awsError:            false,
 		},
 		{
-			name:    "Update machine event failed, reconciler's update failed",
-			machine: machine,
+			name: "Update machine event failed, reconciler's update failed",
 			operation: func(actuator *Actuator, machine *machinev1.Machine) {
 				actuator.Update(context.TODO(), machine)
 			},
-			event:    "Warning FailedCreate error",
-			awsError: true,
+			event:               "aws-actuator-testing-machine: reconciler failed to Update machine: AWS error",
+			invalidMachineScope: false,
+			awsError:            true,
 		},
 		{
-			name:    "Update machine event succeed",
-			machine: machine,
+			name: "Update machine event succeed and only one event is created",
 			operation: func(actuator *Actuator, machine *machinev1.Machine) {
 				actuator.Update(context.TODO(), machine)
+				actuator.Update(context.TODO(), machine)
```

Comment:
The machine is created and reconciled, then we call update over itself. why is this expecting an event?  Isn't this what this PR is trying to avoid?

Reply:
We expect the event to be created only if the machine's resource version was changed after `Patch()`. In this case first `Update()` triggers an event and the second one not.

Reply:
oh right I missed we are doing two calls